doc/manual/pragmas.txt

```diff
@@ -1087,3 +1087,68 @@ In the above example, providing the -d flag causes the symbol
 ``FooBar`` to be overwritten at compile time, printing out 42. If the
 ``-d:FooBar=42`` were to be omitted, the default value of 5 would be
 used.
+
+
+User custom pragmas 
+-------------------
+It is possible to define custom typed pragmas. Custom pragmas do not effect 
+code generation directly, but their presence can be detected by macros.
+Custom pragmas are defined using pragma ``pragma``:
+
+.. code-block:: nim
+  template dbTable(name: string, table_space: string = nil) {.pragma.}
+  template dbKey(name: string = nil, primary_key: bool = false) {.pragma.}
+  template dbForeignKey(t: typedesc) {.pragma.}
+  template dbIgnore {.pragma.}
+
+Consider stylized example of possible Object Relation Mapping (ORM) implementation:
+.. code-block:: nim
+  const tblspace {.strdefine.} = "dev" # switch for dev, test and prod environments 
+
+  type
+    User {.dbTable: ["users", tblspace].} = object
+      id {.dbKey: [primary_key = true].}: int
+      name {.dbKey: "full_name".}: string
+      is_cached {.dbIgnore.}: bool
+      age: int
+
+    UserProfile {.dbTable: ["profiles", tblspace].} = object
+      id {.dbKey: [primary_key = true].}: int
+      user_id {.dbForeignKey: User.}: int
+      read_access: bool
+      write_access: bool
+      admin_acess: bool
+
+In this example custom pragmas are used to describe how Nim objects are 
+mapped to the schema of the relational database. Custom pragmas can have 
+zero, one or more arguments. In order to pass multiple arguments enclose
+them into brackets `[]` as for the rest of the pragmas. If you need to pass
```

Comment:
This needs to be done with ``()`` instead, other pragmas already use tuple syntax and it makes more sense than the array syntax.

Reply:
I have done it in the latest commit, however it required a minor parser change to make `pass by name` work. I was getting nkAsgn nodes instead of nkExprEqExpr